src/test/resources/refaster/CollectionIsEmptyRecipe.java

```diff
@@ -0,0 +1,118 @@
+ * limitations under the License.
+ */
+package foo;
+
+import org.jspecify.annotations.NullMarked;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.*;
+import org.openrewrite.java.template.Primitive;
+import org.openrewrite.java.template.function.*;
+import org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor;
+import org.openrewrite.java.tree.*;
+
+import javax.annotation.Generated;
+import java.util.*;
+
+import static org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor.EmbeddingOption.*;
+
+/**
+ * OpenRewrite recipe created for Refaster template {@code CollectionIsEmpty}.
+ */
+@SuppressWarnings("all")
+@NullMarked
+@Generated("org.openrewrite.java.template.processor.RefasterTemplateProcessor")
+public class CollectionIsEmptyRecipe extends Recipe {
+
+    /**
+     * Instantiates a new instance.
+     */
+    public CollectionIsEmptyRecipe() {}
+
+    @Override
+    public String getDisplayName() {
+        //language=markdown
+        return "Refaster template `CollectionIsEmpty`";
+    }
+
+    @Override
+    public String getDescription() {
+        //language=markdown
+        return "Recipe created for the following Refaster template:\n```java\nclass CollectionIsEmpty<T> {\n    \n    @BeforeTemplate()\n    boolean before(Collection<T> collection) {\n        return Refaster.anyOf(collection.size() == 0, collection.size() <= 0, collection.size() < 1, collection.stream().findAny().isEmpty(), collection.stream().findFirst().isEmpty());\n    }\n    \n    @AfterTemplate()\n    boolean after(Collection<T> collection) {\n        return collection.isEmpty();\n    }\n}\n```\n.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+            final JavaTemplate before$0 = JavaTemplate
+                    .builder("#{collection:any(java.util.Collection<T>)}.size() == 0")
+                    .build();
+            final JavaTemplate before$1 = JavaTemplate
+                    .builder("#{collection:any(java.util.Collection<T>)}.size() <= 0")
+                    .build();
+            final JavaTemplate before$2 = JavaTemplate
+                    .builder("#{collection:any(java.util.Collection<T>)}.size() < 1")
+                    .build();
+                    .build();
+            final JavaTemplate after = JavaTemplate
+                    .builder("#{collection:any(java.util.Collection<T>)}.isEmpty()")
+                    .build();
+
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) { // TODO visitBinary
```

Comment:
This generated method is wrong; possibly due to the Refaster.anyOf on the input.